### DISCUSSION TITLE
Add support for modding via Patch Operations

### DISCUSSION THREAD

**Discussion by leonard84** [FakeFishGames/Barotrauma#8430] [Ideas and suggestions]
Rimworld has powerful XML Modding support via [PatchOperations](https://rimworldwiki.com/wiki/Modding_Tutorials/PatchOperations), allowing mod authors to write modifications for the game files, instead of having to copy and modifying it. This makes it possible for multiple mods to modify the same item without conflict, as long as they don't modify the same value. Furthermore, it makes mods more stable as the item will retain all non-modified properties from its base version, thus being properly updated. Another advantage is, that mods can also add something to other mods without having to know them explicitly, just by specifying a general xpath.

Having something similar for Batorauma would make modding easier and more powerful.

There is a [third-party-tool](https://github.com/Jlobblet/Barotrauma-Mod-Generator) doing something in this direction, but it is more limited and only works for a single mod.

(Transferred from #8422)

**Comment by SashaIndustries** [comment]
Please add this!

**Reply by bobingabout** [reply]
Agreed, this should be added.
I was linked to this page when asking if such a thing exists.

I'm a long time modder from Factorio, overriding an existing item with your own version is considered a Sin in Factorio modding, you're strongly encouraged to not do that, and instead insert a line into an existing entity. In fact they recommend if you're creating a new entity, that you instead use a command to clone an existing one, rather than copy paste the code, so any changes to the base item in the vanilla game from an update ripple though to your derived entity.

So the fact that redefining the item is the only option in Barotrauma modding appaulles me.

**Reply by shangjiaxuan** [reply]
@bobingabout I had a naive implementation for Items and Characters that have this functionality, and created a pr a long time ago. The pr is not complete in that it only supports Items and Characters, and implementing for others will need extra work in PrefabCollection to allow manual defered initialization.

Pr is here:
https://github.com/Regalis11/Barotrauma/pull/9779

This patch integrated within client side lua is available on workshop:
https://steamcommunity.com/workshop/filedetails/?id=2856758496
